pkg/controller/keystone/keystone_controller.go

```diff
@@ -550,3 +563,19 @@ func (r *ReconcileKeystone) listKeystonePods(keystoneName string) (*core.PodList
 	}
 	return pods, nil
 }
+
+func newKeystoneService(cr *contrail.Keystone) *core.Service {
+	return &core.Service{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      cr.Name + "-service",
+			Namespace: cr.Namespace,
+			Labels:    map[string]string{"service": cr.Name},
+		},
+		Spec: core.ServiceSpec{
```

Comment:
setting Spec should be inside the mutate function of CreateOrUpdate since it can be overwritten by Get method called inside CreateOrUpdate. In description of this method is said that desired state of object should be set by mutate function